framework/resource/resource.go

```diff
@@ -40,6 +40,18 @@ type Resource struct {
 	destroyFunc         interface{}
 	platform            string
 	categoryDisplayHint pb.ResourceCategoryDisplayHint
+	statusFunc          interface{}
+
+	statusResp *StatusResponse
+}
+
+// StatusResponse is a container type that holds the resources status reports. A
+// resource can have 1 status response containing zero to many individual status
+// reports, depending on the resource. An example would be a K8s deployment
+// resource returning a single status containing a StatusReport_Resource for
+// each pod it currently tracks
+type StatusResponse struct {
+	Reports []*pb.StatusReport_Resource
```

Comment:
Last thing - Can we rename this to Resources? I think it'll be more clear to plugin devs this way, that resourceManager gives them status report "resources", and they're responsible for collating them into a `report`.
```suggestion
	Resources []*pb.StatusReport_Resource
```

Reply:
Sure; renamed in b63e2c0